src/main/java/io/kestra/storage/gcs/GcsStorage.java

```diff
@@ -38,13 +43,24 @@ private Storage client() {
     }
 
     private BlobId blob(String tenantId, URI uri) {
+        String path = getPath(tenantId, uri);
+        return blob(path);
+    }
+
+    @NotNull
+    private BlobId blob(String path) {
+        return BlobId.of(this.config.getBucket(), path);
+    }
+
+    @NotNull
+    private String getPath(String tenantId, URI uri) {
```

Comment:
I don't think we need it as directory do not exist in gcs .. in the name should not refer to parent directory. I'll just add a test for it to gain more confidence.

Reply:
It has been kept in all cloud storage implementations anyway even if it's not a real issue to keep exception messages aligned between implementations.
So it may be a good idea to add it back there (or to remove it in other implementations except the local storage).